src/ansys/geometry/core/designer/body.py

```diff
@@ -877,10 +882,15 @@ def color(self) -> str:  # noqa: D102
                 # Fetch color from the server if it's not cached
                 color_response = self._bodies_stub.GetColor(EntityIdentifier(id=self._id))
                 if color_response.color:
-                    self._color = mcolors.to_hex(color_response.color)
+                    self._color = mcolors.to_hex(color_response.color, keep_alpha=True)
 
         return self._color
 
+    @property
+    def opacity(self) -> float:  # noqa: D102
+        opacity_hex = self._color[7:]
+        return int(opacity_hex, 16) / 255 if opacity_hex else 1
```

Comment:
What about an opacity setter? You could pass in a value between 0 and 1 (float) and do the conversion behind the scenes to the full RGBA value.